jackmix: add to nixpkgs

kampfschlaefer opened the pull request:
My first real nix package.

One question (also in the source): jack needs libuuid, thus all packages compiling against it also need libuuid (for the headers). So shouldn't libuuid be a propagatedBuildInput for jack[12] instead of a buildInput?

Thanks to @garbas and the devops-sprint in Halle;-)

kampfschlaefer committed:
jackmix: init at version 0.5.2

Add jackmix to the nix pkgs.

davidak commented:
You should use the same nickname as you have on Github so it is easy to mention you when someone edited a package you maintain.

also you have a typo in the homepage:
`http://www.anoldarts.de/jackmix/`
`http://www.arnoldarts.de/jackmix/`

the package builds successfully on Linux (NixOS) x86_64 :+1: 

thanks for contributing!

lethalman commented:
Thanks. Commit messages should follow the contributing guidelines: http://nixos.org/nixpkgs/manual/#idm140737318577088

lethalman commented:
Yes, uuid must be propagatedBuildInputs of jack1.

kampfschlaefer commented:
thanks, is it okay to do the jack-related changes in this PR? Or should I create a new PR to add uuid to jack1's propagatedBuildInputs and check all the other pkgs depending on jack1?

lethalman commented:
@kampfschlaefer it's ok to do it in this PR, preferably a different commit. But I don't see any other package depending on jack1?

garbas commented:
in this PR is ok, since it is related work

On Fri, Oct 2, 2015 at 12:13 PM, Arnold Krille <notifications@github.com>
wrote:

> thanks, is it okay to do the jack-related changes in this PR? Or should I
> create a new PR to add uuid to jack1's propagatedBuildInputs and check all
> the other pkgs depending on jack1?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/10152#issuecomment-144982902>.
>



-- 
Rok Garbas
http://www.garbas.si
rok@garbas.si

kampfschlaefer committed:
jackaudio: Add libuuid as propagatedBuildInput

The jack/jack.h includes uuid/uuid.h, therefox libuuid should be a propagated input for anything using jack.

kampfschlaefer committed:
jackmix: Build jackmix in two flavours

Flavours are:
- *jackmix* using jack2 (default)
- *jackmix_jack1* using jack1

kampfschlaefer commented:
Locally compiled jackmix and jackmix-jack1 (both worked).

lethalman commented:
Commit messages are still not following guidelines....

kampfschlaefer commented:
@lethalman Oky, so your "should" is more of a "have to"…

lethalman commented on pkgs/applications/audio/jackmix/default.nix:
```diff
@@ -0,0 +1,37 @@
+    sha256 = "18f5v7g66mgarhs476frvayhch7fy4nyjf2xivixc061ipn0m82j";
+  };
+
+  jackdep = if jack == "2" then pkgs.libjack2 else pkgs.jack1;
```
This is very bad. Let people pass the jack derivation directly as input.

lethalman commented:
Built locally thanks.

lethalman commented:
I've just seen it.... did you add uuid to propagated build inputs for jack2 too?

lethalman commented:
Removed in b3d1184693b85e63daed01f305097b70a5dc66ba, jack2 does not need and does not have to propagate uuid.

kampfschlaefer commented:
@bjornfor sounds good. But I think pulling in pkgs is from the examples in the wiki. I will probably change that with the next cleanup.

@lethalman as far as I have seen during my local compiles, the jack/jack.h of jack2 also includes uuid/uuid.h. At least both jack-versions should share the header.

lethalman commented:
Weird I did grep jack2 and didn't find any uuid.h. Will check again.

On Sat, Oct 3, 2015 at 12:06 AM, Arnold Krille <notifications@github.com>
wrote:

> @bjornfor <https://github.com/bjornfor> sounds good. But I think pulling
> in pkgs is from the examples in the wiki. I will probably change that with
> the next cleanup.
>
> @lethalman <https://github.com/lethalman> as far as I have seen during my
> local compiles, the jack/jack.h of jack2 also includes uuid/uuid.h. At
> least both jack-versions should share the header.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/10152#issuecomment-145167282>.
>



-- 
NixOS Linux <http://nixos.org>

kampfschlaefer commented:
Hi @lethalman,

I checked again and found that with jack2, its actually lash including the uuid.h in its own header, so the propagated build input should be there. I opened #10549 for this, maybe you can take a look?
